setup.py

```diff
@@ -40,6 +40,7 @@ def get_version() -> str:
     "pytest-cov",
     "datasets",
     "soundfile",
+    "fsspec",
```

Comment:
It should be added to its own `extras` so that it can be installed via `pip install huggingface_hub[fsspec]` or be added as a requirement for a downstream library requiring it